### PR TITLE
warpd: init at 1.2.2

### PR DESCRIPTION
**Pull request by renesat**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/warpd/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    libXft
+    libXfixes
+    libXtst
+    libX11
+    libXext
+  ];
+
+  installPhase = ''
+    mkdir -p $out/{bin,share/man/man1}
+    install -m644 warpd.1.gz $out/share/man/man1/
```
Comment (fabaff) [inline pkgs/tools/X11/warpd/default.nix:26]: `installManPage` which is part of `installShellFiles` could help here.

Comment (renesat) [reply pkgs/tools/X11/warpd/default.nix]: OK, add `installManPage`

**Comment by totoroot** [comment]
Result of `nixpkgs-review pr 171304` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>warpd</li>
  </ul>
</details>

**Comment by poita66** [comment]
@renesat Thanks for packaging warpd.

FYI I noticed that there's also [pull/166584](https://github.com/NixOS/nixpkgs/pull/166584) for the same repo.
How would we go about consolidating these PRs?

Also, see [my comment](https://github.com/NixOS/nixpkgs/pull/166584#issuecomment-1127203023) about X11/Wayland versions as I'd love your input

**Comment by renesat** [comment]
@poita66. I agree with rename package to warp-x11/warp-wayland. I can also try build package for wayland, but without test.

And I didn't notice that he has releases, and I'll change PR for the latest one ([v1.2.1](https://github.com/rvaiya/warpd/releases/tag/v1.2.2))  🤷‍♂️

**Comment by renesat** [comment]
@poita66 I update version, add wayland build and create 2 packages: warpd-wayland, warpd-x11.

File: pkgs/applications/accessibility/warpd/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{ waylandBuild ? false, lib, fetchurl, stdenv, xorg, installShellFiles, cairo
+, libxkbcommon, wayland }:
+
+stdenv.mkDerivation rec {
+  name = "warpd";
+  version = "1.2.2";
+
+  src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/warpd/default.nix:8]: Please use fetchFromGitHub.

File: pkgs/applications/accessibility/warpd/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{ waylandBuild ? false, lib, fetchurl, stdenv, xorg, installShellFiles, cairo
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/warpd/default.nix:1]: Can we build wayland and x at the same time?

File: pkgs/applications/accessibility/warpd/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  };
+
+  buildInputs = if waylandBuild then [
+    wayland
+    cairo
+    libxkbcommon
+  ] else
+    (with xorg; [ libXi libXinerama libXft libXfixes libXtst libX11 libXext ]);
+
+  PLATFORM = if waylandBuild then "wayland" else "";
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/warpd/default.nix:21]:
```suggestion
  PLATFORM = lib.optionalString waylandBuild "wayland";
```

File: pkgs/applications/accessibility/warpd/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    installManPage warpd.1.gz
+    install -m755 bin/warpd $out/bin/
+  '';
+
+  meta = with lib; {
+    description = "A modal keyboard-driven virtual pointer";
+    homepage = "https://github.com/rvaiya/warpd";
+    platforms = platforms.linux ++ platforms.darwin;
+    maintainers = with maintainers; [ renesat ];
+    mainProgram = "warpd";
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/warpd/default.nix:39]:
```suggestion
```
That is the default

File: pkgs/applications/accessibility/warpd/default.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  installPhase = ''
+    mkdir -p $out/bin
+    installManPage warpd.1.gz
+    install -m755 bin/warpd $out/bin/
+  '';
+
+  meta = with lib; {
+    description = "A modal keyboard-driven virtual pointer";
+    homepage = "https://github.com/rvaiya/warpd";
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/warpd/default.nix:37]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -33959,6 +33959,9 @@ with pkgs;
 
   keynav = callPackage ../tools/X11/keynav { };
 
+  warpd-x11 = callPackage ../applications/accessibility/warpd { waylandBuild = false; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:33962]:
```suggestion
  warpd = callPackage ../applications/accessibility/warpd { };
```

File: pkgs/applications/accessibility/warpd/default.nix (hunk 3 above)
Comment (renesat) [reply pkgs/applications/accessibility/warpd/default.nix]: No. Only wayland or X11. This is controlled only by the PLATFORM environment variable - "wayland" or "macos". If other - X11.

**Comment by renesat** [comment]
@SuperSandro2000 Fix and add PLATFORM value for macos

File: pkgs/applications/accessibility/warpd/default.nix [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+  buildInputs = if waylandBuild then [
+    wayland
+    cairo
+    libxkbcommon
+  ] else
+    (with xorg; [ libXi libXinerama libXft libXfixes libXtst libX11 libXext ]);
+
+  PLATFORM = if (waylandBuild && stdenv.isDarwin) then
+    builtins.abort "Wayland not support on Darwin!"
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/warpd/default.nix:23]:
```suggestion
    throw "Wayland not support on Darwin!"
```

**Review by McSinyx — APPROVED** [review]
Binary is working as expected on X and man page is OK.

**Comment by renesat** [comment]
@SuperSandro2000 I think I fix It.

File: pkgs/applications/accessibility/warpd/default.nix [hunk 9]
```diff
@@ -0,0 +1,48 @@
+{ waylandBuild ? false, lib, fetchFromGitHub, stdenv, xorg, installShellFiles
+, cairo, libxkbcommon, wayland, darwin }:
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/warpd/default.nix:2]:
```suggestion
, cairo, libxkbcommon, wayland, ApplicationServices, Cocoa }:
```

File: pkgs/top-level/all-packages.nix [hunk 10]
```diff
@@ -33959,6 +33959,9 @@ with pkgs;
 
   keynav = callPackage ../tools/X11/keynav { };
 
+  warpd = callPackage ../applications/accessibility/warpd { };
+  warpd-wayland = callPackage ../applications/accessibility/warpd { waylandBuild = true; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:33963]:
```suggestion
  warpd = callPackage ../applications/accessibility/warpd {
    darwin.apple_sdk.frameworks; [ ApplicationServices Cocoa ]
  };

  warpd-wayland = warpd.override { waylandBuild = true; };
```

File: pkgs/applications/accessibility/warpd/default.nix [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-0Pm6sqJ1ekZoUshKu9JnDFj5dBVRRdAoO8gyJnRMcS4=";
+  };
+
+  buildInputs = if waylandBuild then [
+    wayland
+    cairo
+    libxkbcommon
+  ] else if stdenv.isDarwin then
+    (with darwin.apple_sdk.frameworks; [ ApplicationServices Cocoa ])
```
Comment (SuperSandro2000) [inline pkgs/applications/accessibility/warpd/default.nix:20]:
```suggestion
    [ ApplicationServices Cocoa ]
```

**Pull request closed without merging**